geppetto.js/geppetto-ui/src/3d-canvas/threeDEngine/MeshFactory.js

```diff
@@ -504,14 +517,43 @@ export default class MeshFactory {
     return scene;
   }
 
-  parseBase64 (str){
+  async loadThreeGLTFModelFromNode (node) {
+    const loader = new GLTFLoader();
+    const dracoLoader = new DRACOLoader();
+    dracoLoader.setDecoderPath('https://raw.githubusercontent.com/mrdoob/three.js/dev/examples/js/libs/draco/gltf/');
```

Comment:
This will load the js decoding libraries every time we load a node.
We should reuse the loader.

E.g. initialize loader in the React state. It should lazy load the libraries automatically.
At least by looking into the impl of `DRACOLoader` constructor and `setDecoderPath` it doesn't fetch anything

Reply:
I see that threejs has also a global [Cache](https://threejs.org/docs/#api/en/loaders/Cache) but it's disabled by default. If we enable it it would fetch the libraries only once. This is useful across multiple Canvases

For a single Canvas, I would still rather maintain one loader per Canvas since the Loader uses a workerPool that only makes sense if the loader is reused.

Reply:
MeshFactory is a helper class, not a react component. With the latest changes, I initialized the loaders in the constructor, rather than in the React state as you mentioned. Let me know if that's what you had in mind.